### DISCUSSION TITLE
Docker Healthcheck

### DISCUSSION THREAD

**Discussion by stanthewizzard** [gotson/komga#1464] [Q&A]
Hello all

Is there a way to have an healthcheck of komga ?

Edit:

healthcheck:
      test:
        - CMD
        - bash
        - -c
        - exec 5<>/dev/tcp/127.0.0.1/25600
      interval: 10s
      timeout: 30s
      retries: 10

Thanks

**Comment by gotson** [comment]
check the `/actuator/health` endpoint

**Reply by stanthewizzard** [reply]
ok 
"status":"UP"
Could you explain how to add it to docker compose ?
no curl no wget in the image 
thanks

**Reply by gotson** [reply]
>  Could you explain how to add it to docker compose ?

that's out of scope of Komga

**Reply by stanthewizzard** [reply]
in your docker image there is no curl nor wget nor the ability to install it.
How to add the healthcheck for docker in this case ?
Thank you again for your kind help

**Reply by gotson** [reply]
i have no idea what you are trying to do 🤷🏻

**Reply by stanthewizzard** [reply]
Docker health check is a feature provided as part of the Docker container platform. It lets you assess and report the health of a running container based on the application it hosts. You can use a health check to identify if the application running in the Docker container is functioning correctly or not.

A Docker health check operates by executing a command within the container at regular intervals. This command, defined by the user, could be anything that represents the health of the application within the container. If the command executes successfully, the container is considered healthy. If it fails, the container is flagged as unhealthy.

This feature is important because it helps you keep track of the operational status of containerized applications. This can improve availability and reliability and also aid in early problem detection, failover, scalability, and auditing.

Within docker compose today I use:
healthcheck:
test:
- CMD
- bash
- -c
- exec 5<>/dev/tcp/127.0.0.1/25600
interval: 10s
timeout: 30s
retries: 10

When it fails
Container is unhealthy (and container autoheal can restart it)

**Reply by gotson** [reply]
i know what a docker health check is, thanks.

curl/wget is not required in the docker image as of now, adding it would just add bloat. You could create your own docker image based on the Komga one.

**Reply by stanthewizzard** [reply]
ok ok :)

ok I'll try
